### PR TITLE
Suite list implemented on data context, and suite list also implemented in CLI.

### PR DESCRIPTION
**Pull request by talagluck**
Testing also added for both the data context `suite_list` and the CLI `suite list`.

### CODE REVIEW

File: tests/cli/test_suite.py [hunk 1]
```diff
@@ -676,3 +677,68 @@ def test_suite_edit_one_datasources_no_generator_with_no_additional_args(
...
+    assert ("No expectation suites available" in result.output)
+
+    assert_no_logging_messages_or_tracebacks(caplog, result)
+
+def test_suite_list_with_one_suite(
+    caplog, empty_data_context
+):
+    project_dir = empty_data_context.root_directory
+    context = DataContext(project_dir)
+    context.create_expectation_suite("a_warning")
```
Comment (jcampbell) [inline tests/cli/test_suite.py]: A minor preference, but I'd like to ensure our test code uses "." as the delimiter for suite names in all examples/tests to reinforce the intended hierarchical naming.

**Review by jcampbell — DISMISSED** [review]

File: great_expectations/cli/suite.py [hunk 2]
```diff
@@ -228,3 +228,35 @@ def suite_new(suite, directory, view, batch_kwargs):
...
+        return
+
+    if len(suite_names) == 1:
+        cli_message("1 expectation suite available:")
+
+    if len(suite_names) > 1:
+        cli_message(f"{len(suite_names)} expectation suites available:")
+
+    for name in suite_names:
+        cli_message(f"\t{name}")
```
Comment (Aylr) [inline great_expectations/cli/suite.py]: I can't believe I'm the one advocating for a python 2 string fix...
```suggestion
        cli_message("\t{}".format(name))
```

File: great_expectations/cli/suite.py [hunk 3]
```diff
@@ -228,3 +228,35 @@ def suite_new(suite, directory, view, batch_kwargs):
...
+    suite_names = context.list_expectation_suite_names()
+    if len(suite_names) == 0:
+        cli_message("No expectation suites available")
+        return
+
+    if len(suite_names) == 1:
+        cli_message("1 expectation suite available:")
+
+    if len(suite_names) > 1:
+        cli_message(f"{len(suite_names)} expectation suites available:")
```
Comment (Aylr) [inline great_expectations/cli/suite.py]:
```suggestion
         cli_message("{} expectation suites available:".format(len(suite_names)))
```

**Review by Aylr — DISMISSED** [review]

File: great_expectations/cli/suite.py (hunk 2 above)
Comment (Aylr) [reply great_expectations/cli/suite.py]: DONT DO IT

File: tests/data_context/test_data_context.py [hunk 4]
```diff
@@ -1058,3 +1058,24 @@ def test_load_config_variables_file(basic_data_context_config, tmp_path_factory)
...
+    assert observed == []
+
+def test_list_expectation_suite_with_one_suite(titanic_data_context):
+    titanic_data_context.create_expectation_suite('warning')
+    observed = titanic_data_context.list_expectation_suite_names()
+    assert isinstance(observed, list)
+    assert observed == ['warning']
+
+def test_list_expectation_suite_with_multiple_suites(titanic_data_context):
+    titanic_data_context.create_expectation_suite('a,warning')
```
Comment (Aylr) [inline tests/data_context/test_data_context.py]:
```suggestion
    titanic_data_context.create_expectation_suite('a.warning')
```

**Review by Aylr — APPROVED** [review]
rad

**Pull request merged**